stable_plugins/visualization/file_types/qasm_visualization.py

```diff
@@ -275,57 +276,61 @@ def get_requirements(self) -> str:
 TASK_LOGGER = get_task_logger(__name__)
 
 
-@CELERY.task(
-    name=f"{QasmVisualization.instance.identifier}.create_circuit_image", bind=True
-)
-def create_circuit_image(self, db_id: int) -> str:
-    TASK_LOGGER.info(f"Starting new demo task with db id '{db_id}'")
-    task_data: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
-
-    if task_data is None:
-        msg = f"Could not load task data with id {db_id} to read parameters!"
-        TASK_LOGGER.error(msg)
-        raise KeyError(msg)
-
-    circuit_url = task_data.data.get("circuit_url", None)
-    url_hash = hashlib.sha256(circuit_url.encode("utf-8")).hexdigest()
-    PluginState.set_value(
-        QasmVisualization.instance.identifier, f"timeout_{url_hash}", 2, commit=True
-    )
+@CELERY.task(name=f"{QasmVisualization.instance.identifier}.generate_image", bind=True)
+def generate_image(self, url: str, hash: str) -> str:
+    TASK_LOGGER.info(f"Generating image for circuit {url}...")
     try:
-        with open_url(circuit_url) as qasm_response:
+        with open_url(url) as qasm_response:
             circuit_qasm = qasm_response.text
     except HTTPError:
-        PluginState.set_value(
+        TASK_LOGGER.error(f"Invalid circuit URL: {url}")
+        DataBlob.set_value(
             QasmVisualization.instance.identifier,
-            url_hash,
+            hash,
             "",
             commit=True,
         )
-        PluginState.delete_value(
-            QasmVisualization.instance.identifier, f"timeout_{url_hash}", commit=True
-        )
+        PluginState.delete_value(QasmVisualization.instance.identifier, hash, commit=True)
         return "Invalid circuit URL!"
+
     circuit = QuantumCircuit.from_qasm_str(circuit_qasm)
     fig = circuit.draw(output="mpl")
-    with SpooledTemporaryFile() as output:
-        fig.savefig(output, format="png")
-        file = STORE.persist_task_result(
-            db_id,
-            output,
-            f"circuit_{url_hash}.png",
-            "image/png",
-            "image/png",
-        )
-        PluginState.set_value(
-            QasmVisualization.instance.identifier,
-            url_hash,
-            file.file_storage_data,
-            commit=True,
-        )
-
-    PluginState.delete_value(
-        QasmVisualization.instance.identifier, f"timeout_{url_hash}", commit=True
+    figfile = BytesIO()
+    fig.savefig(figfile, format="png")
```

Comment:
This should be save for images, but a SpooledTemporaryFile has the advantage, that it evicts the content from ram to disk if it gets too large automatically. (Again this is probably only relevant for stupidly large images of stupidly large circuits, so no need to fix)